tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteApplication.cs

```diff
@@ -201,11 +202,17 @@ public string UniqueFolderName
 
         public string DestinationNewRelicExtensionsDirectoryPath { get { return Path.Combine(DestinationNewRelicHomeDirectoryPath, "extensions"); } }
 
+        private AgentLogFile _agentLogFile;
+
         public AgentLogFile AgentLog
         {
             get
             {
-                return new AgentLogFile(DestinationNewRelicLogFileDirectoryPath, AgentLogFileName, Timing.TimeToConnect);
+                if (_agentLogFile == null)
```

Comment:
consider using newer C# syntax
```return _agnetLogFile ??= new AgentLogFile(estinationNewRelicLogFileDirectoryPath, AgentLogFileName, Timing.TimeToConnect);```

Reply:
if the _agentLogFile is null and a new one is instantiated, _agentLogFile needs to be assigned that newly instantiated object

Reply:
There is a problem with it. In the event where a test app fails to run, the test runner will catch it and retry, the directory that contains the test app and agent for the retry will be in a different directory than what the `_agentLogFile` is referring.

Reply:
the directory that contains the test app and agent for the retry is defined (partially) by the UniqueFolderName, which is also cached in a backing field, so will not change on the retry. Therefore, the cached AgentLog file name will be accurate.

However, while testing this behavior on a retry, I found some other (legacy) issues in the RemoteApplicationFixture.Initialize() method which caused the retry mechanism to not work. Mostly, that the retry flag called appIsExercisedNormally gets reset based on the exit code of the application, which is often a valid 0 even if there were other exceptions from running the test. I will update the PR description to elaborate.

Reply:
resolved Jason's syntax request